packages/video_player/video_player/lib/video_player.dart

```diff
@@ -378,6 +379,9 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
 
   Future<ClosedCaptionFile>? _closedCaptionFileFuture;
   ClosedCaptionFile? _closedCaptionFile;
+  ClosedCaptionFile? _previousClosedCaptionFile;
```

Comment:
This is unnecessarily duplicating the memory requirements any time the file changes, and also does an extremely expensive compare for no reason every time. We know when the file is reset, without a compare, because there is an API to set it. We don't need extra state or checks to determine that.